sdk/src/main/java/ly/count/android/sdk/ModuleViews.java

```diff
@@ -467,26 +467,31 @@ void updateOrientation(int newOrientation) {
         updateOrientation(newOrientation, false);
     }
 
-    void pauseRunningViewsAndSend() {
-        L.d("[ModuleViews] pauseRunningViewsAndSend, going to the background and pausing");
+    void stopRunningViewsAndSend() {
+        L.d("[ModuleViews] stopRunningViewsAndSend, going to the background and pausing");
         for (Map.Entry<String, ViewData> entry : viewDataMap.entrySet()) {
             ViewData vd = entry.getValue();
+            vd.willStartAgain = true;
 
             if (vd.viewStartTimeSeconds > 0) {
                 //if the view is running
-                pauseViewWithIDInternal(vd.viewID, true);
+                stopViewWithIDInternal(vd.viewID, null);
             }
         }
     }
 
-    void resumeAutoPausedViews() {
-        L.d("[ModuleViews] resumeAutoPausedViews, going to the foreground and resuming");
-        for (Map.Entry<String, ViewData> entry : viewDataMap.entrySet()) {
-            ViewData vd = entry.getValue();
-
-            if (vd.isAutoPaused) {
-                //if the view was automatically paused, resume it
-                resumeViewWithIDInternal(vd.viewID);
+    void startAutoStoppedViews() {
+        L.d("[ModuleViews] startAutoStoppedViews, app is coming back to the foreground, starting views that were paused");
+
+        Iterator<Map.Entry<String, ViewData>> iterator = viewDataMap.entrySet().iterator();
+        while (iterator.hasNext()) {
+            Map.Entry<String, ViewData> value = iterator.next();
+            iterator.remove();
+            ViewData vd = value.getValue();
+            if (vd.willStartAgain) {
+                //if the view is paused
+                iterator.remove();
+                startViewInternal(vd.viewName, vd.viewSegmentation, vd.isAutoStoppedView);
```

Comment:
viewSegmentation is not needed here, its the segmentation provided after starting a view.